backend/overstroomik_service/geoserver.py

```diff
@@ -0,0 +1,152 @@
+
+                            elif f_id.startswith(settings.LAYER_EVACUATION_PERCENTAGE):
+                                data_item["evacuation_percentage"] = properties.get(
+                                    settings.FIELD_EVACUATION_PERCENTAGE, None)
+
+                            elif f_id.startswith(settings.LAYER_FLOOD_TYPE):
+                                data_item["flood_type"] = properties.get(
+                                    settings.FIELD_FLOOD_TYPE, None)
+
+                        else:
+                            status: ERROR_GEOS_NO_SMAP
+                else:
+                    status = ERROR_GEOS_NO_RESP
+
+        return status, Data(**data_item)
+
+    def get_api_url_from_rd(self,
+                            rd_x: float,
+                            rd_y: float,
+                            geoserver_url: Optional[str] = settings.GEOSERVER_URL,
+                            layers: [str] = settings.GEOSERVER_LAYER):
+        """
+        Create the get-feature-info link.
+        :param rd_x: x coordinate in meters, geocoded location longitude transformed into EPSG:28992
+        :param rd_y: y coordinate in meters, geocoded location latitude transformed into EPSG:28992
+        :param geoserver_url: link to the geoserver (example: http://geoserver:8080/geoserver)
+        :param layers: group layer with the expected data  (example: overstroomik:Overstroomik_data)
+        """
+
+        # api url template
+        api_get_feature_info = f"{geoserver_url}/overstroomik/wms?SERVICE=WMS&VERSION=1.1.1"\
+            f"&REQUEST=GetFeatureInfo&INFO_FORMAT=application/json&SRS=EPSG:28992&FEATURE_COUNT=50"\
+            f"&LAYERS={layers}&QUERY_LAYERS={layers}"
+
+        # bounding box
+        bounding_box_rd = {
+            "min_x": float(634),
+            "min_y": float(306594),
+            "max_x": float(284300),
+            "max_y": float(636981)
+        }
+
+        # test the input coordinate is in layer-extend
+        test_coordinate = rd_x >= bounding_box_rd.get(
+            "min_x") and rd_x <= bounding_box_rd.get(
+            "max_x") and rd_y >= bounding_box_rd.get(
+            "min_y") and rd_y <= bounding_box_rd.get("max_y")
+
+        # calculate the height and width
+        height = bounding_box_rd.get("max_y") - bounding_box_rd.get("min_y")
+        width = bounding_box_rd.get("max_x") - bounding_box_rd.get("min_x")
+
+        # calculate x/y relative to the extent
+        ddx = rd_x - bounding_box_rd.get("min_x")
+        ddy = rd_y - bounding_box_rd.get("min_y")
+
+        # calculate then x/y in pixels
+        x = width * ddx / width
+        y = height - (height * ddy / height)
+
+        # bounding box format
+        bounding_box = f"{bounding_box_rd['min_x']},{bounding_box_rd['min_y']}"\
+            f",{bounding_box_rd['max_x']},{bounding_box_rd['max_y']}"
+
+        return {
+            "valid": test_coordinate,
+            "height": height,
```

Comment:
The object this returns is mostly unused. Only the state and url are used, maybe pass these back as tuple?